src/main/kotlin/com/jetbrains/micropython/repl/DeviceCommsManager.kt

```diff
@@ -0,0 +1,97 @@
+package com.jetbrains.micropython.repl
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.project.Project
+import com.jediterm.terminal.TtyConnector
+import com.jetbrains.micropython.settings.MicroPythonFacet
+import com.jetbrains.micropython.settings.firstMicroPythonFacet
+import com.pty4j.PtyProcess
+import org.jetbrains.plugins.terminal.LocalTerminalDirectRunner
+import java.lang.ref.WeakReference
+
+sealed class CommsEvent {
+    class ProcessStarted(val ttyConnector: TtyConnector) : CommsEvent()
+    class ProcessCreationFailed(val reason: String) : CommsEvent()
+    object ProcessDestroyed : CommsEvent()
+}
+
+interface CommsEventObserver {
+    fun onCommsEvent(event: CommsEvent)
+}
+
+@Service
+class DeviceCommsManager(project: Project) {
```

Comment:
I'd name it `MicroPythonReplManager`. Most classes start with `MicroPython*` so they are easy to find via Shift-Shift or Cmd-O. It's only about REPL, so not all communications, if I'm not mistaken.